app/models/student.rb

```diff
@@ -99,11 +99,9 @@ def absences_overall
   end
 
   def solos(filtered_course = nil)
-    if filtered_course
-      attendance_records.where(pair_ids: [], ignore: nil).where("date between ? and ?", filtered_course.try(:start_date), filtered_course.try(:end_date)).select { |ar| !ar.date.friday? }.count
-    else
-      attendance_records.where(pair_ids: [], ignore: nil).select { |ar| !ar.date.friday? }.count
-    end
+    solo_records = attendance_records.includes(:pairings).where(pairings: {id: nil})
```

Comment:
This line isn't used when `filtered_course` is provided, right?

Reply:
It is used by the line that builds filtered_records, regardless of the presence of filtered_course. I refactored that method (student#solos) because I thought this way of writing it was easier to read, but now I'm not totally convinced.

Reply:
Oh never mind I just didn't read the ternary statement correctly. I would suggest not using the ternary operator for a line that long.

Reply:
Makes sense. Yeah I'll replace with a simple if else.